xcel_itron2mqtt/xcelEndpoint.py

```diff
@@ -32,14 +36,18 @@ def __init__(self, session: requests.session, mqtt_client: mqtt.Client,
         # Setup the rest of what we need for this endpoint
         self.mqtt_send_config()
 
+    @retry(stop=stop_after_attempt(15),
+           wait=wait_exponential(multiplier=1, min=1, max=15),
+           before_sleep=before_sleep_log(logger, logging.WARNING),
+           reraise=True)
     def query_endpoint(self) -> str:
         """
         Sends a request to the given endpoint associated with the 
         object instance
 
         Returns: str in XML format of the meter's response
         """
-        x = self.requests_session.get(self.url, verify=False, timeout=4.0)
+        x = self.requests_session.get(self.url, verify=False)
```

Comment:
If I remember correctly, when I ran the request function without a timeout, it could cause the whole process to hang for an extremely long time if no reply was received. Does the retry decorator cover this potential issue?

Reply:
Or it will retry when the timeout occurs. We may need to increase the timeout beyond 4s if we need to keep it in. Maybe something like 10s? At least on my Ubiquiti equipment, if the meter has fallen asleep, it takes it about 10s to start responding again. We could just use the retry logic to handle that scenario